app/src/main/java/com/hover/stax/utils/UIHelper.kt

```diff
@@ -44,12 +44,17 @@ object UIHelper {
         s.show()
     }
 
-    fun flashMessage(context: Context, message: String?) {
+    fun flashMessage(context: Context, message: String, isAppError: Boolean = false) {
         Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+        AnalyticsUtil.logAnalyticsEvent(message, context)
+        if(isAppError) AnalyticsUtil.logErrorAndReportToFirebase(context.getString(R.string.toast_err_tag), message, null)
     }
 
-    fun flashMessage(context: Context, messageRes: Int) {
-        Toast.makeText(context, context.getString(messageRes), Toast.LENGTH_SHORT).show()
+    fun flashMessage(context: Context, messageRes: Int, isAppError: Boolean = false) {
```

Comment:
Doing this changes what the method is meant to do. Let's separate logging from showing toasts. In keeping with SOLID principles, it'll help during testing.

Reply:
@alexosugo I partly disagree. We always want flash messages to be logged, so the analytics call should be here. The method name should change to make this clear `flashAndReportMessage` or something, but we specifically don't want to have to remember to log the message every time we show a flash. 

That said the boolean arg is a clear violation of SOLID, so prob want a second method instead for errors `flashAndReportError` or something.

Reply:
@davkutalek Agreed. Having a separate method makes sense.